etherman/etherman.go

```diff
@@ -79,6 +79,10 @@ type Client struct {
 	Matic                 *matic.Matic
 	SCAddresses           []common.Address
 
+	// readOnly if `true` the client won't have an account.
+	// Check this field before trying to access the `auth` field.
+	readOnly bool
+
```

Comment:
My suggestion is to remove this configuration and if the key is not provided, the methods requiring the auth should fail, returning an error message.

it's ok to have a variable to control if it is read-only or not, but this variable should be related to the keystore config and not a new one.

in summary, if we have a keystore file, we instantiate it and provide it to etherman, so etherman knows it can perform actions that require authorization. If we don't have the keystore, we don't instantiate it and provide nil to etherman, so etherman knows it cannot perform actions that require authorization.

Reply:
thanks @tclemos, it makes totally sense. It should be better now 346ac5d85620d86e5991e6c3ae744f409262f51a, WDYT?